PWGUD/Tasks/upcForward.cxx

```diff
@@ -27,35 +29,43 @@ using namespace std;
-     {"hPt", ";#it{p_{t}}, GeV/c;", {HistType::kTH1D, {{500, 0., 5.}}}},
-     {"hPtsingle_muons", ";#it{p_{t}}, GeV/c;", {HistType::kTH1D, {{500, 0., 5.}}}},
-     {"hPx", ";#it{p_{x}}, GeV/c;", {HistType::kTH1D, {{500, -5., 5.}}}},
-     {"hPy", ";#it{p_{y}}, GeV/c;", {HistType::kTH1D, {{500, -5., 5.}}}},
-     {"hPz", ";#it{p_{z}}, GeV/c;", {HistType::kTH1D, {{500, -5., 5.}}}},
-     {"hRap", ";#it{y},..;", {HistType::kTH1D, {{500, -10., 10.}}}},
-     {"hEta", ";#it{y},..;", {HistType::kTH1D, {{500, -10., 10.}}}},
-     {"hCharge", ";#it{charge},..;", {HistType::kTH1D, {{500, -10., 10.}}}},
-     {"hSelectionCounter", ";#it{Selection},..;", {HistType::kTH1I, {{30, 0., 30.}}}},
-     {"hPhi", ";#it{#Phi},;", {HistType::kTH1D, {{500, -6., 6.}}}}}};
+  // defining histograms using histogram registry
+  HistogramRegistry registry{"registry", {}, OutputObjHandlingPolicy::AnalysisObject};
 
   void init(o2::framework::InitContext&)
   {
-    //TString SelectionCuts[8] = {"NoSelection", "CMup11and10Trigger","V0Selection","FDSelection", "twotracks", "oppositecharge", "-2.5<Eta<-4", "Pt<1"};
 
-    /* // commenting out this part now as histogram registry do not seem to suppor the binlabel using AxisSpec() or any other method
+    auto hSelectionCounter = registry.add<TH1>("hSelectionCounter", "hSelectionCounter;;NEvents", HistType::kTH1I, {{30, 0., 30.}});
+
+    // THIS IS THE SELECTION CUTS FOR NOW. We will add further cuts
+    TString SelectionCuts[8] = {"NoSelection", "CMup11and10Trigger", "V0Selection", "FDSelection", "twotracks", "oppositecharge", "-2.5<Eta<-4", "Pt<1"};
+    // now we can set BinLabel in histogram Registry
     for (int i = 0; i < 6; i++) {
-      registry.GetXaxis().(HIST("hSelectionCounter",SetBinLabel(i + 1, SelectionCuts[i].Data())));
-    }*/
+      hSelectionCounter->GetXaxis()->SetBinLabel(i + 1, SelectionCuts[i].Data());
+    }
+
+    registry.add("hMass", "Mass of Mother;#it{m_{#mu#mu}}, GeV/c^{2};", kTH1D, {{500, 0., 10.}});
+    registry.add("hPt", "Pt of Mother;#it{p_{t}}, GeV/c;", kTH1D, {{500, 0., 5.}});
+    registry.add("hPtsingle_muons", "Pt of Daughters;#it{p_{t}}, GeV/c;", kTH1D, {{500, 0., 5.}});
+    registry.add("hPx", "Px;#it{P_{x}}, GeV/c;", kTH1D, {{500, -5., 5.}});
+    registry.add("hPy", "Py;#it{P_{y}}, GeV/c;", kTH1D, {{500, -5., 5.}});
+    registry.add("hPz", "Pz;#it{P_{z}}, GeV/c;", kTH1D, {{500, -5., 5.}});
+    registry.add("hRap", "Rapidity of Mother;#it{y};", kTH1D, {{500, -10., 10.}});
+    registry.add("hEta", "Eta;#it{#eta};", kTH1D, {{500, -10., 10.}});
+    registry.add("hCharge", "Charge;#it{charge};", kTH1D, {{500, -10., 10.}});
+    registry.add("hPhi", "Phi;#it{#Phi};", kTH1D, {{500, -6., 6.}});
+
+    Configurable<float> etalow{"etalow", -4.0f, ""};   //
+    Configurable<float> etahigh{"etahigh", -2.5f, ""}; //
+    Filter etaFilter_a = aod::fwdtrack::eta > etalow;
+    Filter etaFilter_b = aod::fwdtrack::eta < etahigh;
```

Comment:
You should combine in a single Filter expression these two lines; it should be more efficient

Reply:
Hi Ionut. I think they are combined automatically and it should have the same efficiency.
@aalkin can you comment?

Reply:
If there is such an optimization then its fine by me

Reply:
@iarsene @jgrosseo I already merged them into one.

Reply:
Yes, indeed, compatible filters will be joined with `and` automatically, but merging them also reduces the number of data members of the task so you can add something else if needs be.